app/javascript/onboarding/__tests__/Onboarding.test.jsx

```diff
@@ -58,85 +70,104 @@ describe('<Onboarding />', () => {
       profile_image_url: 'dev.jpg',
     },
   ]);
-
   const dataUser = JSON.stringify({
     followed_tag_names: ['javascript'],
   });
 
   describe('IntroSlide', () => {
     let onboardingSlides;
+
     beforeEach(() => {
-      document.body.setAttribute('data-user', null);
-      onboardingSlides = deep(<Onboarding />);
+      onboardingSlides = initializeSlides(0);
     });
 
     test('renders properly', () => {
       expect(onboardingSlides).toMatchSnapshot();
     });
 
-    test('should move to the next slide upon clicking the next button', () => {
+    test('should advance', () => {
       onboardingSlides.find('.next-button').simulate('click');
       expect(onboardingSlides.state().currentSlide).toBe(1);
     });
   });
 
   describe('EmailTermsConditionsForm', () => {
     let onboardingSlides;
+    const codeOfConductCheckEvent = {
+      target: {
+        value: 'checked_code_of_conduct',
+        name: 'checked_code_of_conduct',
+      },
+    };
+    const termsAndConditionsCheckEvent = {
+      target: {
+        value: 'checked_terms_and_conditions',
+        name: 'checked_terms_and_conditions',
+      },
+    };
+    const updateCodeOfConduct = () => {
+      onboardingSlides
+        .find('#checked_code_of_conduct')
+        .simulate('change', codeOfConductCheckEvent);
+    };
+    const updateTermsAndConditions = () => {
+      onboardingSlides
+        .find('#checked_terms_and_conditions')
+        .simulate('change', termsAndConditionsCheckEvent);
+    };
+
     beforeEach(() => {
-      document.body.setAttribute('data-user', dataUser);
-      onboardingSlides = deep(<Onboarding />);
-      onboardingSlides.setState({ currentSlide: 1 });
+      onboardingSlides = initializeSlides(1, dataUser);
     });
 
     test('renders properly', () => {
       expect(onboardingSlides).toMatchSnapshot();
     });
 
-    test('should not move if code of conduct is not agreed to', () => {
+    // Arguably this test is actually just testing the Preact framework
+    // but for the sake of detecting a regression I refactoring it instead
+    // of removing it (@jacobherrington)
```

Comment:
I think there's a typo: `I refactoring it` instead should be `I am refactoring it`, also you don't need to add your name, `git blame` will reveal your identity :D

Reply:
I use git blame for that, but I like to include my name in case people are less comfortable with git :)